tapestry.js

```diff
@@ -155,8 +163,26 @@ function init() {
     // 2. SIZE AND SCALE THE TAPESTRY AND SVG TO FIT WELL
     //---------------------------------------------------
 
-    // Do it now
+    // establish initial positioning before the startforce()
+    // this so 'getNodeDimensions' is operational at this stage
+    for (var index in dataset.nodes) {
+        if (dataset.nodes[index].x == null) {
+            dataset.nodes[index].x = getBrowserWidth()*0.9;
+        }
+        if (dataset.nodes[index].y == null) {
+            dataset.nodes[index].y = getBrowserHeight()*0.9;
+        }
+    }
+
+    // do it now
     updateTapestrySize();
+
+    $(window).resize(function() {
+        var browserRatio = getBrowserWidth()/getBrowserHeight();
```

Comment:
Why don't you just call `updateTapestrySize()` instead on window resize?

Reply:
If I recall correctly `updateTapestrySize()` was what screwed up the whole autolayout and window resizing, making it too small

Reply:
Have you tested it? I think I changed it to updateTapestrySize() on my local and it was working fine. If you test it and it works fine, then I'd rather use that since it's more high-level and easier to read.

Reply:
corrected @ 177